frontend/e2e-test/test/e2e/pages/employee/units/unit-page.ts

```diff
@@ -126,6 +127,31 @@ export default class UnitPage {
     '[data-qa="mobile-device-name-input"]'
   )
   readonly mobileDevicePairingDoneBtn = Selector('[data-qa="modal-okBtn"]')
+
+  readonly employeeOptions = Selector('[id^="react-select-2-option"]')
+
+  async addMobileDevice(): Promise<{ pairingId: UUID; deviceId: UUID | null }> {
+    await t.expect(this.mobileDevicesTableRows.exists).notOk()
+    await t.click(this.mobileDevicesStartPairingBtn)
+    await t.expect(this.pairingModalTitle.exists).ok()
+    await t.expect(this.mobileDevicesChallengeKey.exists).ok()
+    const challengeKey = await this.mobileDevicesChallengeKey.textContent
+
+    const res = await postPairingChallenge(challengeKey)
+    await t.expect(this.mobileDevicesResponseKeyInput.exists).ok()
+    if (res.responseKey) {
+      await t.typeText(this.mobileDevicesResponseKeyInput, res.responseKey)
+    }
+    await t.expect(this.mobileDevicesNameInput.exists).ok()
+
+    const pairingId = res.id
+    const deviceId = await this.mobileDevicesNameInput.getAttribute('id')
+
+    await t.typeText(this.mobileDevicesNameInput, 'testphone')
+    await t.click(this.mobileDevicePairingDoneBtn)
+
+    return { pairingId, deviceId }
```

Comment:
I personally don't see the value in doing assertions like this `await t.expect(this.mobileDevicesResponseKeyInput.exists).ok()` since the `await t.typeText(this.mobileDevicesResponseKeyInput, res.responseKey)` kind of does the same assertion as well. 

Also, if a missing `res.responseKey` means that the test can't succeed, should you just throw an error? Now the test just skips inputting the value silently.

Reply:
Removed the extra `exists).ok` checks and added a `throw new Error` when not receiving the responseKey.